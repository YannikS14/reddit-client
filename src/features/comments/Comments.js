import React, { useState, useEffect } from 'react';
import axios from 'axios';
import Comment from '../comment/Comment';
import Skeleton from 'react-loading-skeleton';

export default function Comments({ postSubreddit, postId }) {
  const [comments, setComments] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const [hasErrors, setHasErrors] = useState(false);

  useEffect(() => {
    const fetchComments = async () => {
      try {
        setIsLoading(true);
        const response = await axios.get(
          `https://www.reddit.com/r/${postSubreddit}/comments/${postId}.json`,
        );
        setComments(response.data[1].data.children);
      } catch (e) {
        setHasErrors(true);
      } finally {
        setIsLoading(false);
      }
    };

    fetchComments();
  }, [postSubreddit, postId]);

  return (
    <div id="post-comments" className="mt-2">
      {isLoading && <Skeleton count={3} />}
      {!hasErrors &&
        !isLoading &&
        comments.slice(0, 10).map((comment) => {
          return (
            <Comment
              comment={comment}
              key={comment.data.id}
              isLoading={isLoading}
            />
          );
        })}
    </div>
  );
}
