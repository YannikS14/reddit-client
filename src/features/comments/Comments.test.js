import renderer from 'react-test-renderer';
import Comments from './Comments';

describe('Comments component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Comments />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
