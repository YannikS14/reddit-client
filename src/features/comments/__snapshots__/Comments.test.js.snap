// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments component matches the snapshot 1`] = `
<div
  className="mt-2"
  data-testid="comments-wrapper"
  id="post-comments"
/>
`;
